lmfit/confidence.py

```diff
@@ -183,11 +183,13 @@ def __init__(self, minimizer, result, p_names=None, prob_func=None,
         # check that there are at least 2 true variables!
         # check that all stderrs are sensible (including not None or NaN)
         nvars = 0
-        for par in self.fit_params:
+        for par in self.params.values():
             if par.vary:
                 nvars += 1
+        for par in self.fit_params:
             try:
-                if not (par.vary and par.stderr > 0.0):
+                if not (par.vary and par.stderr is not None and
+                        par.stderr is not np.nan):
                     raise MinimizerException(CONF_ERR_STDERR)
             except TypeError:
                 raise MinimizerException(CONF_ERR_STDERR)
```

Comment:
it seems that the logic here should/could be re-written; if I understand it correctly it should be in pseudo-code:
"if parameter varies and parameter is not a positive, finite number --> raise MinimizerException"

If you test now for ```None``` and ```np.nan``` is it then ever going to give a ```TypeError```?

Reply:
You are right, the TypeError check is now not necessary now.  I can I either revert to the initial form or remove the try.

Reply:
let's not revert to the initial form... I would like to get rid of the ```not``` in the if-statement and rewrite the condition so that it works

Reply:
can we re-write the ```if-statement``` to:  ```if par.vary and (par.stderr is None or par.stderr is np.nan)```? At least that's what I think it is intended to be and it still passes the tests.